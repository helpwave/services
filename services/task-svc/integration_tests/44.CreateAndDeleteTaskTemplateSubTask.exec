#!/bin/bash

echo "Invalid id"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"taskTemplateId": "40ee9afbf1b81c16b3ab871", "name": "Some Name"}' \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/CreateTaskTemplateSubTask 2>&1 || echo ""

echo "id does not exist"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"taskTemplateId": "51aa9af4-9037-4d56-bf1b-81c16b3ab871", "name": "Some Name"}' \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/CreateTaskTemplateSubTask 2>&1 || echo ""

echo "Create"
id=$(grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"taskTemplateId": "40ee9af4-9037-4d56-bf1b-81c16b3ab871", "name": "Some Name"}' \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/CreateTaskTemplateSubTask | jq -r ".id")

echo "delete it again"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d "{\"id\": \"$id\"}" \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/DeleteTaskTemplateSubTask

echo "there should be no change"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"created_by": "73c0d265-3c9a-410a-8eaf-0174fc57f421"}' \
	localhost:3001 \
	services.task_svc.v1.TaskTemplateService/GetAllTaskTemplatesByCreator
