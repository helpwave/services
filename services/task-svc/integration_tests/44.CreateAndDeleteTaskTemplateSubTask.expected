Invalid id
ERROR:
  Code: Unknown
  Message: invalid UUID length: 23

id does not exist
ERROR:
  Code: Internal
  Message: ERROR: insert or update on table "task_template_subtasks" violates foreign key constraint "task_template_subtasks_task_template_id_fkey" (SQLSTATE 23503)

Create
delete it again
{}
there should be no change
{
  "templates": [
    {
      "id": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd",
      "name": "Reattachment of Right Foot, Open Approach",
      "description": "Administration of diphtheria-tetanus-pertussis, combined",
      "isPublic": true,
      "subtasks": [
        {
          "id": "86617de9-4012-4ee3-9ecb-1fc0c9e7ab85",
          "taskTemplateId": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd",
          "name": "Reposition R Metacarpocarp Jt w Ext Fix, Perc Endo"
        },
        {
          "id": "5a712e12-c2a0-472e-827a-0e754a396378",
          "taskTemplateId": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd",
          "name": "Change Traction Apparatus on Right Lower Extremity"
        },
        {
          "id": "a77d20cb-1954-4e4f-baed-a5e28c0b620e",
          "taskTemplateId": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd",
          "name": "Destruction of Left Trunk Tendon, Perc Endo Approach"
        },
        {
          "id": "972830e0-28f8-4e28-8187-e63e11514915",
          "taskTemplateId": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd"
        },
        {
          "id": "c6e6bde8-ca25-4a8f-80b5-cc6697d8f523",
          "taskTemplateId": "526ff1c0-f373-4d2b-bcd8-d355f8223cfd",
          "name": "Removal of Radioactive Element from L Lung, Extern Approach"
        }
      ]
    },
    {
      "id": "58a8c60d-3be5-497a-9afa-a53b85ef10d8",
      "name": "Supplement Left Elbow Joint with Nonaut Sub, Open Approach",
      "isPublic": true,
      "subtasks": [
        {
          "id": "cc00d575-5c88-48ab-91b4-a147bae5c7bb",
          "taskTemplateId": "58a8c60d-3be5-497a-9afa-a53b85ef10d8",
          "name": "Endocrine System, Drainage"
        }
      ]
    },
    {
      "id": "47d91496-1351-48be-9f62-09cd819cf9fc",
      "name": "Removal of Nonaut Sub from Bladder, Perc Approach",
      "isPublic": true
    },
    {
      "id": "defe5b74-913d-49ce-8e43-f2c603e3a297",
      "name": "Dilate R Popl Art w 4+ Intralum Dev, Perc Endo",
      "description": "Microscopic examination of specimen from skin and other integument, culture and sensitivity",
      "isPublic": true
    },
    {
      "id": "40ee9af4-9037-4d56-bf1b-81c16b3ab871",
      "name": "Pharmacotherapy for Substance Abuse Treatment, Naloxone",
      "isPublic": true
    }
  ]
}
